proposals/reward-proposal-v2-reward-distribution.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Epoch Reward Distribution
+---
+
+## Problem
+
+In the `Epoch Reward Calculation V2`, we proposed a new method to handle epoch
+rewards. This is the second part of that proposal - `Reward Distribution`.
+
+## Proposed Solutions
+
+Similar to [Partitioned Inflationary Rewards
+Distribution](https://github.com/solana-labs/solana/blob/master/docs/src/proposals/partitioned-inflationary-rewards-distribution.md),
+the rewards is distributed over `M` slots.
+
+To address the concern for the lack of on-chain proof for reward distribution, a
+special system instruction, `DistributeReward`, will be introduced to distribute
+the rewards from the `EpochRewardProof` system account. This instruction takes
+the target reward account, the amount to distribute and the merkle proof for the
+distribution. The new instruction will be executed. When executing this
+instruction, it first verifies the supplied merkle proof against the proof
+stored in in `EpochRewardProof`. If the verification passed, it will transfer
+the reward from `EpochRewardProof` account to the target reward account.
+
+To spread out the reward distribution, for each slot, a fixed number (`K`) of
+stake/vote accounts's rewards are distributed. The leaders, within `M`
+slot-height, will be responsible for distributing the rewards. The leader will
+inject `K` `DistributeReward` instructions as transactions into the block and
+propagate those transaction down to the network. All nodes on the network will
+execute those transactions and distribute the rewards. In this way, all reward
+distribution will be recoded as transactions in the first `M` blocks after
+reward computation.
+
+### Challenges and Difficulties
+
+The following are a few questions and challenges of the `V2` reward
+distribution approach for further discussion.
+
+1. What are the incentives for the leader to pack reward redeem transactions?
```

Comment:
Yeah. 
To reject the block produced by the leader, we could make all validators check for the *expected* reward transactions when replaying during reward interval. 

The idea of adding inflation economic for redeeming rewards is very interesting.

Reply:
Updated the proposal to include both ideas.

Reply:
This definitely need to be addressed in the main text of this proposal.
It could also be an on-chain requirement that the validators should process at least `n` many pending rewards distributions during their leader slots or otherwise the slot is rejected. Don't know the drawbacks with this though.